reference-implementation/lib/transform-stream.js

```diff
@@ -45,10 +47,16 @@ function TransformStreamEnqueueToReadable(transformStream, chunk) {
     throw transformStream._storedError;
   }
 
-  const backpressure = controller.desiredSize <= 0;
+  const maybeBackpressure = controller.desiredSize <= 0;
 
-  if (transformStream._readableBackpressure !== backpressure) {
-    TransformStreamSetBackpressure(transformStream, backpressure);
+  if (maybeBackpressure === true && transformStream._backpressure === false) {
+    // This allows pull() again. When desiredSize is 0, it's possible that a pull() will happen immediately (but
+    // asynchronously) after this because of pending read()s and set _backpressure back to false.
+    //
+    // We don't have to worry about that such a pull() may happen inside the enqueue() call, and therefore is not
```

Comment:
How about:

> If pull() could be called from inside enqueue(), then this logic would be wrong. This cannot happen because there is always a promise pending from start() or pull() when _backpressure is false.